internal/factsengine/gatherers/hostsfile.go

```diff
@@ -69,7 +69,11 @@ func (s *HostsFileGatherer) Gather(factsRequests []entities.FactRequest) ([]enti
 
 	for _, factReq := range factsRequests {
 		var fact entities.Fact
-
+		if factReq.Argument == "" {
```

Comment:
This needs to be double checked, it would probably be `nil`

Reply:
I checked it manually by not providing the `Argument` field in the fact request, and it seems that in such case, `factReq.Argument == ""` (in fact, the tests are passing). But I'll look again and figure out if it can be `nil`  with other inputs)

Reply:
Shame on me! Forgot that go indeed has "" as null value for strings.. my bad :sweat_smile: